src/main/java/hu/blackbelt/structured/map/proxy/MapProxy.java

```diff
@@ -706,6 +708,80 @@ private boolean invokeEquals(Object proxy, Object[] args) throws InvocationTarge
         return false;
     }
 
+    private void invokeAdd(Method m, Object[] args) throws ExecutionException {
+        if (params.isImmutable()) {
+            throw new IllegalStateException("Could not call set on immutable object");
+        }
+        String attrName = Character.toLowerCase(m.getName().charAt(5)) + m.getName().substring(6);
+        AttributeInfo attributeInfo = typeInfoCache.get(clazz).get(attrName);
+        Object values = internal.get(getKeyName(clazz, attrName));
+
+        ArrayList<Object> valuesArray = (values instanceof Collection<?>)
+                ? valuesArray = new ArrayList<>((Collection) values)
+                : new ArrayList<>();
+
+        if (args[0] instanceof Object[]) {
+            Object[] valuesForAdd = (Object[]) args[0];
+            for (Object value : valuesForAdd) {
+                if (attributeInfo != null && attributeInfo.isComposite() && value instanceof MapHolder) {
+                   if (attributeInfo.getPropertyType().isInterface()) {
+                       MapHolder proxy = (MapHolder) MapProxy.builder(attributeInfo.getPropertyType())
+                               .withParams(params)
+                               .withMap(((MapHolder) value).toMap())
+                               .newInstance();
+                       proxy.toMap().entrySet().forEach(e -> {
+                           internal.put(e.getKey(), e.getValue());
+                       });
+                   } else {
+                       throw new IllegalArgumentException(String.format("The attribute %s in %s is not interface. The @Embedded attributes type have to be interface.", attrName, clazz.getName()));
```

Comment:
"not an interface", "has to be"

Reply:
[49a63fa](https://github.com/BlackBeltTechnology/structured-map-proxy/pull/82/commits/49a63faed3d96604466f669b0e028e0f30578d8b)